internal/machinery/machineconfig.go

```diff
@@ -0,0 +1,28 @@
+package machinery
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	machineconfigapi "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NodesUpgrading determines if worker nodes are currently upgrading by comparing
+// MachineCount and UpdatedMachineCount
+func (m *machinery) IsUpgrading(c client.Client, nodeType string, logger logr.Logger) (bool, error) {
+	configPool := &machineconfigapi.MachineConfigPool{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: nodeType}, configPool)
+	if err != nil {
+		logger.Info("Failed to get %s node configpool object.", nodeType)
```

Comment:
Can we remove the logging? Let the user of the function decide what logging to do.